src/go/configgenerator/route_generator.go

```diff
@@ -356,17 +356,22 @@ func MakeRouteTable(serviceInfo *configinfo.ServiceInfo) ([]*routepb.Route, []*r
 		}
 
 		for _, routeMatcher := range routeMatchers {
-			r := makeRoute(routeMatcher, method)
+			bi := method.BackendInfo
+			useLocalHTTPBackend := !method.IsGRPCPath(routeMatcher.GetPath()) && !configinfo.IsDiscoveryAPI(method.Operation()) && method.HttpBackendInfo != nil
+			if useLocalHTTPBackend {
+				bi = method.HttpBackendInfo
+			}
+			r := makeRoute(routeMatcher, method, useLocalHTTPBackend)
```

Comment:
maybe pass in whole "bi"  instead of bool

Reply:
That was my idea too. But since backendInfo is a private field, I used this way so to avoid expose that to public and change other interfaces. Since this is nit, I will just use the current implementation.